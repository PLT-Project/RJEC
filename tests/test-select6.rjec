func foo(ch1 chan char, ch2 chan int, quit chan bool) {
    for {
        select {
        case ch1 <- 'a':
            prints("sending:");
            printc('a');
        case val2 := <- ch2:
            prints("receiving:");
            printi(val2);
        case q := <- quit:
            printb(q);
            if q {
                prints("quitting...");
            }
            return;
        }
    }
}

func main() {
    ch1 := make(chan char);
    ch2 := make(chan int);
    quit := make(chan bool, 10);
    yeet foo(ch1, ch2, quit);
    for i := 0; i < 5; i = i + 1 {
        printc(<-ch1);
        ch2 <- i;
    }
    quit <- true;
}