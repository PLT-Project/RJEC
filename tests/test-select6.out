sending:
a
a
receiving:
0
sending:
a
a
receiving:
1
sending:
a
a
receiving:
2
sending:
a
a
receiving:
3
sending:
a
a
receiving:
4
true
quitting...